plugins/extratio/init.js

```diff
@@ -219,10 +231,12 @@ rTorrentStub.prototype.setratiorules = function()
 		var rle = plugin.rules[i];
 		var enabled = $("#_rre"+i).prop("checked") ? 1 : 0;
 		var name = $("#_rrn"+i).val();
-		this.content = this.content+"&name="+encodeURIComponent(name)+"&pattern="+encodeURIComponent(rle.pattern)+"&enabled="+enabled+
-		        "&reason="+rle.reason+
-			"&channel="+rle.channel+"&ratio="+rle.ratio+
-			"&no="+rle.no;
+		this.content += "&name="+encodeURIComponent(name) + "&pattern=" + encodeURIComponent(rle.pattern) + 
+			"&enabled=" + enabled + "&reason=" + rle.reason + 
+			"&ratio=" + rle.ratio + "&no="+rle.no;
+		if (thePlugins.isInstalled("throttle")) {
+			this.content += "&channel=" + rle.channel;
+		}
```

Comment:
Please create a variable outside the for-loop to check if the throttle plugin is enabled. If there are 10 rules, we can avoid up to 40 decision structures, by caching the results outside the for-loop.

https://github.com/Novik/ruTorrent/blob/9bf6e15c4864a71ff9e9ca5ea689fd001931279b/js/plugins.js#L75-L79